### PR TITLE
`help` function shouldn't use pager for AliasHelpInfo

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

The `help` function passes the returned content to a pager unless it is of type `HelpInfoShort` (used when multiple matches found).  This includes passing `AliasHelpInfo` which is short enough to not require passing to the pager.  Fix is to simply pass output to pipeline if the result type is `AliasHelpInfo` rather than through the pager.

https://github.com/PowerShell/PowerShell/issues/8541

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/powershell/engine/Help/HelpSystem.Tests.ps1 [hunk 1]
```diff
@@ -360,6 +360,13 @@ Describe "Get-Help should find pattern alias" -Tags "CI" {
         $help.Category | Should -BeExactly "Alias"
         $help.Synopsis | Should -BeExactly "Where-Object"
     }
+
+    It "Help alias should be same as Get-Help alias" {
+        Set-Alias -Name testAlias1 -Value Where-Object
```
Comment (iSazonov) [inline test/powershell/engine/Help/HelpSystem.Tests.ps1]: Please remove the temp alias after the test.

**Review by iSazonov — APPROVED** [review]
With one minor comment.

File: test/powershell/engine/Help/HelpSystem.Tests.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [reply test/powershell/engine/Help/HelpSystem.Tests.ps1]: Existing line 344 removes the test alias

Comment (SteveL-MSFT) [reply test/powershell/engine/Help/HelpSystem.Tests.ps1]: Looking at the code, I'll move the creation of the alias to a BeforeAll statement

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 2]
```diff
@@ -4170,7 +4170,8 @@ .FORWARDHELPCATEGORY Cmdlet
     $help = Get-Help @PSBoundParameters
 
     # If a list of help is returned, don't pipe to more
```
Comment (adityapatwardhan) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: Maybe also add a comment about why 'AliasHelpInfo' is ignored.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**